source/CargoHold.cpp

```diff
@@ -113,9 +113,10 @@ void CargoHold::Save(DataWriter &out) const
 	if(!first)
 		out.EndChild();
 	
+	// Save all outfits, even ones which have only been referred to.
 	bool firstOutfit = true;
 	for(const auto &it : outfits)
-		if(it.second && !it.first->Name().empty())
+		if(it.second)
```

Comment:
The `NameAndWarn` method ensures any outfits that were referred to or loaded have the appropriate name.

Reply:
Yes, except for older savegames where the name was already lost (and the outfit would be loaded with an empty name by NameAndWarn). But I guess you would also like to keep those outfits so that the data can be restored by players by doing a manual edit.